sdk/storage/storage-file-share/src/ShareFileClient.ts

```diff
@@ -1395,31 +1396,36 @@ export class ShareFileClient extends StorageClient {
   // High Level functions
 
   /**
-   * ONLY AVAILABLE IN BROWSERS.
-   *
-   * Uploads a browser Blob/File/ArrayBuffer/ArrayBufferView object to an Azure File.
+   * Uploads a Buffer(Node)/Blob/ArrayBuffer/ArrayBufferView to an Azure File.
    *
-   * @param {Blob | ArrayBuffer | ArrayBufferView} browserData Blob, File, ArrayBuffer or ArrayBufferView
+   * @param {Buffer | Blob | ArrayBuffer | ArrayBufferView} data Buffer(Node), Blob, ArrayBuffer or ArrayBufferView
    * @param {FileParallelUploadOptions} [options]
    * @returns {Promise<void>}
    */
-  public async uploadBrowserData(
-    browserData: Blob | ArrayBuffer | ArrayBufferView,
+  public async uploadData(
+    data: Buffer | Blob | ArrayBuffer | ArrayBufferView,
     options: FileParallelUploadOptions = {}
   ): Promise<void> {
-    const { span, spanOptions } = createSpan(
-      "ShareFileClient-uploadBrowserData",
-      options.tracingOptions
-    );
+    const { span, spanOptions } = createSpan("ShareFileClient-uploadData", options.tracingOptions);
     try {
-      const browserBlob = new Blob([browserData]);
-      return this.uploadSeekableBlob(
-        (offset: number, size: number): Blob => {
-          return browserBlob.slice(offset, offset + size);
-        },
-        browserBlob.size,
-        { ...options, tracingOptions: { ...options!.tracingOptions, spanOptions } }
-      );
+      if (isNode && data instanceof Buffer) {
+        const bufferToStream = new ReadableStreamBuffer();
+        bufferToStream.put(data);
+        bufferToStream.stop();
+        return this.uploadResetableStream(() => bufferToStream, data.byteLength, {
```

Comment:
Does this do the right thing if a user passes a buffer larger than FILE_RANGE_MAX_SIZE or specifies a rangeSize option smaller than the buffer size? IIUC, at that point the stream callback will be called with offset/size pairs and we'll just return the same stream for each. Unless rangeSize matches the stream's highWaterMark, which seems very likely, this feels incorrect...

Reply:
Addressed this by updating the callback with offset and count pair.